syntax = "proto3";

package MQ;

//交换机类型
enum ExchangeType {
  UNKNOWTYPE = 0;
  DIRECT = 1;
  FANOUT = 2;
  TOPIC = 3;
};

//发送方式
enum DeliveryMode {
  UNKNOWMODE = 0;
  UNDURABLE = 1;
  DURABLE = 2;
};

//基本属性
message BasicProperties{
  string id = 1;//消息ID
  DeliveryMode delivery_mode = 2;//消息广播模式
  string routing_key = 3;//消息路由键
};

//有效载荷
message Payload {
  BasicProperties properties = 1;//消息基本属性
  string body = 2;//消息体
  string valid = 3;//是否有效
};

//成员：有效载荷、偏移量、大小
message Message {
  Payload payload = 1;
  uint32 offset = 2;
  uint32 length = 3;
};